socket_wrapper.hpp

```diff
@@ -34,7 +35,11 @@ class Socket
     int                     GetSocketHandle() const {return m_socket_handle;}
     const sockaddr_in*      GetSockAddrStruct() const {return &m_socket_address;}
     const char*             GetIPAddressStr() const;
-    
+    const unsigned short    GetPort() const;
+
+                            operator int();
+    friend bool             operator==(const Socket_sptr&, const Socket_sptr&);
```

Comment:
As `GetSocketHandle` is public class method there is no reason for `==` operators to be this class `friend`s. They can (and I guess should) be simple "free" functions.